sdk/dotnet/Utilities.cs

```diff
@@ -39,7 +40,7 @@ public static InvokeOptions WithVersion(this InvokeOptions? options)
             };
         }
 
-        public static string ExecuteCommand(string command, string[] flags)
+        public static string ExecuteCommand(string command, string[] flags, IDictionary<string, string>? env)
```

Comment:
```suggestion
        public static string ExecuteCommand(string command, string[] flags, IDictionary<string, string> env)
```

I don't see a great reason to make in nullable here, so let's stick to strict type - and then you can remove the null check below.